packages/blade/src/components/Input/OTPInput/OTPInput.tsx

```diff
@@ -99,6 +118,23 @@ const OTPInput = ({
     }
   }, [otpValue, otpLength, name, inputValue, onOTPFilled]);
 
+  useEffect(() => {
```

Comment:
I think this might not be the best approach, prone to breaking. What happens during the initial render during SSR? Would it correctly mask the values for controlled / uncontrolled?

Reply:
I ran a test with controlled input on ssr tests and it does create a snapshot with an input of type password. Is there any other way we can test this?

<img width="655" alt="Screenshot 2023-02-06 at 3 22 08 PM" src="https://user-images.githubusercontent.com/24487274/216940417-68bad27f-1a41-4e7e-85f3-6fa3d657baeb.png">

Reply:
This one seems fine. How about uncontrolled one?

Overall though, this logic still feels complex 😅

Reply:
Let's skip for now since it's hard to validate right now.